gender_novels/analysis/gender_pronoun_freq_analysis.py

```diff
@@ -372,6 +386,156 @@ def sort_every_year(frequency_dict):
+    Creates a bar graph give male/female subject/object frequencies
+    :param she_freq_dict:
+    :param he_freq_dict:
+    :param title: name of the exported file
+    :param x: value of x axis
+    :return:
+    """
+
+    fig, ax = plt.subplots()
+    plt.ylim(0, 1)
+
+    key = []
+
+    for k, v in she_freq_dict.items():
+        key.append(k)
+
+    m_freq = dict_to_list(he_freq_dict)
+    f_freq = dict_to_list(she_freq_dict)
+
+    index = np.arange(len(she_freq_dict.keys()))
+    bar_width = 0.35
+    opacity = 0.4
+
+    ax.bar(index, [1]*len(m_freq), bar_width, alpha=opacity, color='c', label="Male Object")
+    ax.bar(index, m_freq, bar_width, alpha=opacity, color='b', label='Male Subject')
+    ax.bar(index + bar_width, [1]*len(f_freq), bar_width, alpha=opacity, color='#DE8F05',
+           label="Female Object")
+    ax.bar(index + bar_width, f_freq, bar_width, alpha=opacity, color='r', label='Female Subject')
+
+    ax.set_xlabel(x)
+    ax.set_ylabel('Frequency')
+    ax.set_title('Relative Frequencies of Subject to Object Pronouns')
+    ax.set_xticks(index + bar_width / 2)
+    plt.xticks(fontsize=8, rotation=90)
+    ax.set_xticklabels(key)
+    ax.legend()
+
+    fig.tight_layout()
+
+    filepng = "visualizations/" + title + ".png"
+    filepdf = "visualizations/" + title + ".pdf"
+    plt.savefig(filepng, bbox_inches='tight')
+    plt.savefig(filepdf, bbox_inches='tight')
+
+
+
+def run_pronoun_freq(corpus):
+    """
+    Runs a program that uses the instance distance analysis on all novels existing in a given
+    corpus, and outputs the data as graphs
+    :return:
+    """
+
+    all_data = books_pronoun_freq(corpus)
+    
+    gender = freq_by_author_gender(all_data)
+    box_gender_pronoun_freq(gender, my_pal={"male_author": "b", "female_author": "r"},
+                             title="she_freq_by_author_gender", x="Author Gender")
+    date = freq_by_date(all_data)
+    box_gender_pronoun_freq(date, my_pal="Greens", title="she_freq_by_date", x="Years")
+    location = freq_by_location(all_data)
+    box_gender_pronoun_freq(location, my_pal="Blues", title="she_freq_by_location", x="Location")
+
+    sub_v_ob = subject_vs_object_pronoun_freqs(corpus)
+
+    female_gender_sub_v_ob = get_mean(freq_by_author_gender(sub_v_ob[1]))
+    male_gender_sub_v_ob = get_mean(freq_by_author_gender(sub_v_ob[0]))
+    bar_sub_obj_freq(female_gender_sub_v_ob,male_gender_sub_v_ob,"obj_sub_by_auth_gender",
+                     "Author Gender")
+    female_date_sub_v_ob = get_mean(freq_by_date(sub_v_ob[1]))
+    male_date_sub_v_ob = get_mean(freq_by_date(sub_v_ob[0]))
+    bar_sub_obj_freq(female_date_sub_v_ob, male_date_sub_v_ob, "obj_sub_by_year", "Years")
+
+    female_loc_sub_v_ob = get_mean(freq_by_location(sub_v_ob[1]))
+    male_loc_sub_v_ob = get_mean(freq_by_location(sub_v_ob[0]))
+    bar_sub_obj_freq(female_loc_sub_v_ob, male_loc_sub_v_ob, "obk_sub_by_location", "Location")
+
+    # male_tot = dict_to_list(sub_v_ob[0])
```

Comment:
Can you clean up these commented out print statements and then update the PR please?